src/DotNetWorker.Core/Http/HttpRequestData.cs

```diff
@@ -62,5 +64,16 @@ public HttpRequestData(FunctionContext functionContext)
         /// </summary>
         /// <returns>The response instance.</returns>
         public abstract HttpResponseData CreateResponse();
+
+        /// <summary>
+        /// Gets the <see cref="Query"/> containing the request query. 
+        /// </summary>
+        public virtual NameValueCollection Query
+        {
+            get
+            {
+                return HttpUtility.ParseQueryString(Url.Query);
```

Comment:
Every time this prop is accessed, the parse method gets executed. Should we store the results of parse in local variable and return that? (basically moving the logic you have in the GrpcHttpRequestData)?

Reply:
I think that would make sense.